pandas/tests/frame/methods/test_asof.py

```diff
@@ -24,14 +26,18 @@ def test_basic(self, date_range_frame):
         df.loc[15:30, "A"] = np.nan
         dates = date_range("1/1/1990", periods=N * 3, freq="25s")
 
-        result = df.asof(dates)
-        assert result.notna().all(1).all()
-        lb = df.index[14]
-        ub = df.index[30]
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", FutureWarning)
+
+            result = df.asof(dates)
```

Comment:
why is this showing a warning? you need to fix internally so this is not the case

Reply:
because the `self.take(locs, is_copy=False)` in asof will now generate a warning after the deprecation change.
I could change it to default None - but then that generates a SettingWithCopyError due to  ` data.loc[missing] = np.nan` in the block below
```
missing = locs == -1
 data = self.take(locs)
 data.index = where
data.index = where
 data.loc[missing] = np.nan
```
I suppose I could temporarily suppress it by          
```
with pd.option_context('mode.chained_assignment', None):
            data.loc[missing] = np.nan
```
or create a deep copy prior to that step but guess thats not a good approach.

Reply:
you simply step theu the code and out a copy where needed